go-controller/pkg/ovn/egressfirewall.go

```diff
@@ -55,14 +56,22 @@ func newEgressFirewallRule(rawEgressFirewallRule egressfirewallapi.EgressFirewal
 		access: rawEgressFirewallRule.Type,
 	}
 
-	_, _, err := net.ParseCIDR(rawEgressFirewallRule.To.CIDRSelector)
-	if err != nil {
-		return nil, err
-	}
-	efr.to.cidrSelector = rawEgressFirewallRule.To.CIDRSelector
+	if rawEgressFirewallRule.To.DNSName != "" {
+		efr.to.dnsName = rawEgressFirewallRule.To.DNSName
+	} else {
 
+		_, _, err := net.ParseCIDR(rawEgressFirewallRule.To.CIDRSelector)
+		if err != nil {
+			return nil, err
+		}
+		efr.to.cidrSelector = rawEgressFirewallRule.To.CIDRSelector
+	}
 	efr.ports = rawEgressFirewallRule.Ports
 
+	if efr.access == egressfirewallapi.EgressFirewallRuleDeny && len(efr.to.dnsName) != 0 {
+		return nil, fmt.Errorf("DNSName is not allowed in deny rules")
```

Comment:
this was decided after the initial commit, Casey thought it would be more secure to only allow "allow" rules because the case the nodes and masters resolve the dns names differently the admin would not be allowing to a destination without realizing it